paddle/inference/CMakeLists.txt

```diff
@@ -1,12 +1,26 @@
-set(FLUID_CORE_MODULES
-    backward proto_desc paddle_memory executor prune init ${GLOB_OP_LIB})
+set(FLUID_CORE_MODULES proto_desc paddle_memory executor prune init)
 
 cc_library(paddle_fluid_api
     SRCS inference.cc
-    DEPS ${FLUID_CORE_MODULES})
+    DEPS ${FLUID_CORE_MODULES} ${GLOB_OP_LIB})
 
-# Merge all modules into a simgle static library
-cc_library(paddle_fluid DEPS paddle_fluid_api ${FLUID_CORE_MODULES})
+# Merge all modules into a single static library
+cc_library(paddle_fluid DEPS paddle_fluid_api ${FLUID_CORE_MODULES} ${GLOB_OP_LIB})
+
+# Create shared library
+add_library(paddle_fluid_shared SHARED inference.cc)
```

Comment:
shared library不能直接叫`libpaddle_fluid.so`？

Reply:
考虑到两点：
1. 仿照了libpaddle_capi_shared.so
2. 如果名字叫`libpaddle_fluid.so`，那第8行的`cc_library(paddle_fluid ...)`就要换名字了，不然两个库就重名了。

Reply:
一个是静态库，一个是动态库，名字也不能一样？

Reply:
比如，如果按照下面这样写：
```
add_library(paddle_fluid_shared SHARED inference.cc)
add_library(paddle_fluid_shared STATIC inference.cc)
```
cmake会报错：
```
add_library cannot create target "paddle_fluid_shared" because another
  target with the same name already exists.  The existing target is a shared
  library created in source directory
```